cli/src/pcluster/models/cluster.py

```diff
@@ -372,58 +372,46 @@ def __init__(
         self._add_validator(InstanceTypeValidator, priority=1, instance_type=self.instance_type)
 
 
-class ComputeResource(Resource):
-    """Represent the Compute Resource."""
+class BaseComputeResource(Resource):
+    """Represent the base Compute Resource, with the fields in common between all the schedulers."""
 
     def __init__(
         self,
-        instance_type: str,
-        max_count: int = None,
-        min_count: int = None,
-        spot_price: float = None,
         allocation_strategy: str = None,
         simultaneous_multithreading: bool = None,
         efa: Efa = None,
     ):
         super().__init__()
-        self.instance_type = Param(instance_type)
-        self.max_count = Param(max_count, default=10)
-        self.min_count = Param(min_count, default=0)
-        self.spot_price = Param(spot_price)
         self.allocation_strategy = Param(allocation_strategy, default="BEST_FIT")
         self.simultaneous_multithreading = Param(simultaneous_multithreading, default=True)
         self.efa = efa
-        # TODO handle awsbatch
 
 
-class Queue(Resource):
-    """Represent the Queue resource."""
+class BaseQueue(Resource):
+    """Represent the generic Queue resource."""
 
-    def __init__(self, name: str, networking: QueueNetworking, compute_resources: List[ComputeResource]):
+    def __init__(
+        self,
+        name: str,
+        networking: QueueNetworking,
+        storage: Storage = None,
+        compute_type: str = None,
+    ):
         super().__init__()
         self.name = Param(name)
         self.networking = networking
-        self.compute_resources = compute_resources
+        self.storage = storage
+        self.compute_type = Param(compute_type, default="ONDEMAND")
 
 
-class SchedulingSettings(Resource):
-    """Represent the Scheduling configuration."""
+class CommonSchedulingSettings(Resource):
+    """Represent the common scheduler settings."""
```

Comment:
Minor: when should we use Common* or Base* ? And there is another case we don't use prefix at all: _ComputeResourceSchema